pulsar-broker/src/main/java/org/apache/pulsar/broker/service/persistent/PersistentDispatcherMultipleConsumers.java

```diff
@@ -673,7 +673,10 @@ protected int getFirstAvailableConsumerPermits() {
         }
         for (Consumer consumer : consumerList) {
             if (isConsumerAvailable(consumer)) {
-                return consumer.getAvailablePermits();
+                int availablePermits = consumer.getAvailablePermits();
+                if (availablePermits > 0) {
+                    return availablePermits;
+                }
             }
```

Comment:
Is better to change to:

```
if (consumer != null && !consumer.isBlocked()) {
       return consumer.getAvailablePermits();
}
```

So that we don't need to check `availablePermits` twice

Reply:
@codelipenghui Done. PTAL